ocaml/idl/datamodel.ml

```diff
@@ -5249,14 +5251,47 @@ let network_detach_for_vm = call
     ~allowed_roles:_R_VM_POWER_ADMIN
     ()
 
+let network_purpose = Enum ("network_purpose", [
+  "nbd", "Network Block Device service using TLS";
+  "insecure_nbd", "Network Block Device service without integrity or confidentiality: NOT RECOMMENDED";
```

Comment:
I think we should only list here the purposes that we currently use in the code: the top two (and maybe "himn" as well). I think from the user's point of view it would be confusing to have all these new purpose enums that they can add to the network, but they don't do anything.
Also, I think we should add new enums as we need them, because it's easier to add new ones than to remove ones we didn't need.

Reply:
Yes, please add only the nbd ones.

Reply:
I'll do that.